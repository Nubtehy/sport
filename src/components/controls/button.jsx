import React from 'react';
import PropTypes from 'prop-types';
import { withStyle } from 'components/HOCs';
import Button from './button.styled';

const displayName = 'Button';

const propTypes = {
  onClick: PropTypes.func,
  style: PropTypes.objectOf(PropTypes.string),
  children: PropTypes.string,
  name: PropTypes.string,
};

const defaultProps = {
  onClick: undefined,
  style: {},
  children: '',
  name: '',
};

export function ButtonBase({
  name, onClick, style, children,
}) {
  return (
    <Button data-name={name} type="button" onClick={onClick} style={style}>
      {children}
    </Button>
  );
}

ButtonBase.displayName = displayName;
ButtonBase.propTypes = propTypes;
ButtonBase.defaultProps = defaultProps;

export const styleCreator = (theme, { accent }) => ({
  backgroundColor: accent ? theme.colorAccent : theme.colorDefault,
  color: '#ffffff',
  height: '36px',
  padding: '0 18px',
  borderRadius: '18px',
  fontSize: '12px',
  lineHeight: '34px',
});

export default withStyle(ButtonBase, styleCreator);
