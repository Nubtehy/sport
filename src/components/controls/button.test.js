import React from 'react';
import { shallow } from 'enzyme';

import { ButtonBase } from './button';

const testProps = {
  onClick: jest.fn(),
  style: { background: 'blue' },
  name: 'add-form-button',
};

describe('Given an Input component', () => {
  describe('when the component is rendered', () => {
    let component;

    beforeEach(() => {
      component = shallow(<ButtonBase {...testProps} />);
    });

    it('should match the snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
