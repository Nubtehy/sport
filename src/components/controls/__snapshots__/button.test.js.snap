// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an Input component when the component is rendered should match the snapshot 1`] = `
<buttonstyled__Button
  data-name="add-form-button"
  onClick={[MockFunction]}
  style={
    Object {
      "background": "blue",
    }
  }
/>
`;
